### PR TITLE
Add `--library` flag to `compile` command

### PR DESCRIPTION
**Pull request by silvanocerza**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

* **What kind of change does this PR introduce?**

Adds a new flags to and existing command and related gRPC function.

- **What is the current behavior?**

There is currently no way of specifying a single custom library path when calling the `compile` command.

* **What is the new behavior?**

The user can now set the `--library` flag to specify one or more libraries root folder when compiling a sketch.

```
arduino-cli compile -b arduino:avr:uno --library=/path/to/lib /path/to/MySketch
arduino-cli compile -b arduino:avr:uno --library /path/to/lib1 --library /path/to/lib2 /path/to/MySketch
arduino-cli compile -b arduino:avr:uno --library=/path/to/lib1,/path/to/lib2 /path/to/MySketch
```

Libraries specified this way have top priority in case of conflict with identical libraries.

- **Does this PR introduce a breaking change, and is
[titled accordingly](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#breaking)?**

Nope.

* **Other information**:

The docstring for the `--libraries` flag has also been modified to make it clearer and to differentiate it from this new `--library` flag.

The `--libraries` flag expects a path to a folder containing multiple libraries, much like the `<sketchbook>/libraries` folder does.
Assume we have two libraries in these paths:
```
/home/user/workspace/libraries/WiFi101
/home/user/workspace/libraries/AudioZero
```
The `--libraries` flag must be used like so:
```
arduino-cli compile -b arduino:avr:uno --libraries /home/user/workspace/libraries /path/to/MySketch
```

Instead the `--library` flag expects a path to the root folder of a single library, so it must be used like so:
```
arduino-cli compile -b arduino:avr:uno --library /home/user/workspace/libraries/WiFi101 --library /home/user/workspace/libraries/AudioZero /path/to/MySketch
```

---

See [how to contribute](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/)

### CODE REVIEW

File: legacy/builder/types/context.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ type Context struct {
 	BuiltInToolsDirs     paths.PathList
 	BuiltInLibrariesDirs paths.PathList
 	OtherLibrariesDirs   paths.PathList
+	LibrariesDirs        paths.PathList // List of paths pointing to individual libraries root folders
```
Comment (per1234) [inline legacy/builder/types/context.go:70]:
```suggestion
	LibraryDirs        paths.PathList // List of paths pointing to individual library root folders
```

File: legacy/builder/libraries_loader.go [hunk 2]
```diff
@@ -61,6 +61,13 @@ func (s *LibrariesLoader) Run(ctx *types.Context) error {
 		return errors.WithStack(err)
 	}
 
+	for _, dir := range ctx.LibrariesDirs {
```
Comment (per1234) [inline legacy/builder/libraries_loader.go:64]:
```suggestion
	for _, dir := range ctx.LibraryDirs {
```

File: commands/compile/compile.go [hunk 3]
```diff
@@ -137,6 +137,8 @@ func Compile(ctx context.Context, req *rpc.CompileRequest, outStream, errStream
 	builderCtx.OtherLibrariesDirs = paths.NewPathList(req.GetLibraries()...)
 	builderCtx.OtherLibrariesDirs.Add(configuration.LibrariesDir(configuration.Settings))
 
+	builderCtx.LibrariesDirs = paths.NewPathList(req.Library...)
```
Comment (per1234) [inline commands/compile/compile.go:140]:
```suggestion
	builderCtx.LibraryDirs = paths.NewPathList(req.Library...)
```

**Review by per1234 — CHANGES_REQUESTED** [review]

File: cli/compile/compile.go [hunk 4]
```diff
@@ -93,8 +97,10 @@ func NewCommand() *cobra.Command {
 	command.Flags().StringVarP(&port, "port", "p", "", "Upload port, e.g.: COM10 or /dev/ttyACM0")
 	command.Flags().BoolVarP(&verify, "verify", "t", false, "Verify uploaded binary after the upload.")
 	command.Flags().StringVar(&vidPid, "vid-pid", "", "When specified, VID/PID specific build properties are used, if board supports them.")
+	command.Flags().StringSliceVar(&library, "library", []string{},
+		"List of paths to libraries root folders. Libraries set this way have top priority in case of conflicts. Can be used multiple times for different libraries.")
 	command.Flags().StringSliceVar(&libraries, "libraries", []string{},
-		"List of custom libraries paths separated by commas. Or can be used multiple times for multiple libraries paths.")
+		"List of custom libraries dir paths separated by commas. Or can be used multiple times for multiple libraries dir paths.")
```
[the remark this answers is not in the document]
Comment (silvanocerza) [reply cli/compile/compile.go:103]: I use them interchangeably, with this change I wanted to make it clear that `--libraries` must point to a single directory containing multiple libraries folders but am not sure I made it clearer.

I'm open to suggestions.

**Review by ubidefeo — APPROVED** [review]
Tested and working

**Comment by silvanocerza** [comment]
@per1234 if it's ok for you I'll merge.

**Review by per1234 — APPROVED** [review]
Great work! I apologize for being so slow on the re-review.

**Comment by silvanocerza** [comment]
@per1234 no worries. <3

**Pull request merged**